### ISSUE TITLE
Option to force ALGORITHM=INSTANT during ALTER migrations

### ISSUE THREAD

**Issue by strokirk** [adamchainz/django-mysql#1047]
### Description

With many MySQL `ALTER` statements now supporting [Online DDL](https://dev.mysql.com/doc/refman/8.0/en/innodb-online-ddl-operations.html), it would be a great feature to be able to force Django to always add `ALGORITHM=INSTANT` where possible.

While MySQL mostly uses the "best algorithm it can", this is mostly to safeguard against MySQL silently downgrading to another algorithm, which can easily cause site-wide downtime. Better a failed migration than a locked database.

Would this be a reasonable feature to add? I'd be happy to try to make a PR if so, assuming I can understand where to make it.

(Essentially the same feature request as in: https://code.djangoproject.com/ticket/30773)

**Comment by adamchainz** [comment]
I can see how a safeguard feature would be useful.

The implementation I can think of: override the `migrate` command and add an option to enable the feature like `--force-algorithm-instant`. The `handle()` method could add [database instrumentation](https://docs.djangoproject.com/en/stable/topics/db/instrumentation/) to rewrite `ALTER TABLE` queries to insert `ALGORITHM=INSTANT`, similar to the technique [in this post](https://adamj.eu/tech/2020/07/29/backporting-a-django-orm-feature-with-database-instrumentation/).

Overriding `migrate` is a little tricky, as it means that other apps can't also override it without inheriting from Django-MySQL's version. I am not sure many other packages override `migrate` but it's worth bearing in mind.

(Whitenoise [jumps through some hoops](https://github.com/evansd/whitenoise/blob/main/src/whitenoise/runserver_nostatic/management/commands/runserver.py#L16) to inherit from the correct `runserver` command in its override.)

Basically, I'd be happy to review an initial version but I'm on the fence about adding it to the package. With this kind of feature it's best to have another voice saying it's desirable before I commit to maintaining it.
